celltype_annotation/analysis/copykat-rms-exploration.Rmd

```diff
@@ -0,0 +1,224 @@
+We'll assess agreement between the submitter tumor/normal classification and what `copyKat` infers as tumor/normal.
+
+
+## Set Up
+
+```{r}
+suppressPackageStartupMessages({
+  library(SingleCellExperiment)
+  library(copykat) # https://github.com/navinlabcode/copykat
+})
+
+# define output path and SCE file
+output_dir <-  here::here("celltype_annotation", 
+                         "analysis", 
+                         "copykat-results")
+fs::dir_create(output_dir)
+
+copykat_sce <- file.path(output_dir,
+                         glue::glue("{params$library}_copykat_sce.rds"))
+```
+
+In the next chunk, we obtain the _submitter-annotated_ SCE from S3, if it is not already locally present.
+We'll wait to read it in when actually running `copyKat`, mostly for memory reasons.
+
+
+```{r}
+# make local data directory if it doesn't exist
+if (!dir.exists(params$local_data_dir)) {
+  dir.create(params$local_data_dir, recursive = TRUE)
+}
+
+# build path to submitter annotated sce file 
+submitter_annotated_sce_file <- glue::glue("{params$library_id}_processed_celltype.rds")
+
+
+local_submitter_sce_path <- file.path(params$local_data_dir,
+                                      params$sample_id,
+                                      submitter_annotated_sce_file)
+
+# get submitter sce from S3 if not already present locally
+if (!(file.exists(local_submitter_sce_path))) {
+  sync_call <- glue::glue('op run -- aws s3 sync {params$s3_submitter_data_dir} {params$local_data_dir}/{params$sample_id} --exclude "*" --include "{submitter_annotated_sce_file}"')
+  system(sync_call, ignore.stdout = TRUE) 
+}
+```
+
+
+## Run CopyKAT
+
+Now, we are ready to run `copyKat` on the SCE, using the _raw expression matrix_ as described in the [GitHub `README` file](https://github.com/navinlabcode/copykat/blob/master/README.md).
+
+Of note, the `README` includes this caveat about the `"distance"` parameter:
+
+> One struggling observation is that none of one clustering method could fit all datasets. In this version, I add a distance parameters for clustering that include "euclidean" distance and correlational distance, ie. 1-"pearson" and "spearman" similarity. In general, corretional distances tend to favor noisy data, while euclidean distance tends to favor data with larger CN segments.
+
+Therefore we'll compare `"euclidian"`, `"spearman"`, and `"pearson"` values for [this parameter](https://github.com/navinlabcode/copykat/blob/b795ff793522499f814f6ae282aad1aab790902f/R/copykat.R#L251-L255).
+
+
+```{r message=FALSE, warning=FALSE}
+# Only run copykat if the final SCE does not exist
+if (file.exists(copykat_sce)) {
+  
+  sce <- readr::read_rds(copykat_sce)
+  
+} else {
+  
+  # Otherwise, read in initial SCE and run copyKat
+  sce <- readr::read_rds(local_submitter_sce_path)
+    
+  purrr::walk(
+    c("spearman", "pearson", "euclidean"), 
+    \(x) {
```

Comment:
```suggestion
    \(distance_parameter) {
```
This is silly, but with the new `purrr` we can be explicit about what we're looping over instead of using `x`. I feel like it's more helpful to have meaningful names for understanding what's happening.